syne_tune/optimizer/schedulers/neuralbands/neuralband.py

```diff
@@ -0,0 +1,382 @@
+                        l_t_score = sorted(l_t_score, key = lambda x: x[1], reverse=True)
+                        config = l_t_score[0][0]
+                        break
+                while_loop_count += 1      
+                
+                    
+            else:
+                self._searcher_initialized = False
+                self._initialize_searcher_new()
+                config = self.searcher.get_config(**extra_kwargs, trial_id=trial_id)
+                break
+                
+        #------------- Ban ----------------
+        if config is not None:
+            config = cast_config_values(config, self.config_space)
+            config = self._on_config_suggest(config, trial_id, **extra_kwargs)
+            config = TrialSuggestion.start_suggestion(config)
+            
+        return config    
+
+        
+
+    
+
+   
+
+
+    def on_trial_result(self, trial: Trial, result: Dict) -> str:
+        self._check_result(result)
+        trial_id = str(trial.trial_id)
+        debug_log = self.searcher.debug_log
+        trial_decision = SchedulerDecision.CONTINUE
+        if len(result) == 0:
+            # An empty dict should just be skipped
+            if debug_log is not None:
+                logger.info(
+                    f"trial_id {trial_id}: Skipping empty dict received "
+                    "from reporter"
+                )
+        else:
+            
+            # added code
+            # Time since start of experiment
+            time_since_start = self._elapsed_time()
+            do_update = False
+            config = self._preprocess_config(trial.config)
+            cost_and_promotion = (
+                self._cost_attr is not None
+                and self._cost_attr in result
+                and self.does_pause_resume()
+            )
+            if cost_and_promotion:
+                # Trial may have paused/resumed before, so need to add cost
+                # offset from these
+                cost_offset = self._cost_offset.get(trial_id, 0)
+                result[self._total_cost_attr()] = result[self._cost_attr] + cost_offset
+            if trial_id not in self._active_trials:
+                # Trial not in self._active_trials anymore, so must have been
+                # stopped
+                trial_decision = SchedulerDecision.STOP
+                logger.warning(
+                    f"trial_id {trial_id}: Was STOPPED, but receives another "
+                    f"report {result}\nThis report is ignored"
+                )
+            elif not self._active_trials[trial_id]["running"]:
+                # Trial must have been paused before
+                trial_decision = SchedulerDecision.PAUSE
+                logger.warning(
+                    f"trial_id {trial_id}: Was PAUSED, but receives another "
+                    f"report {result}\nThis report is ignored"
+                )
+            else:
+
+                # added code -------------- Ban ------------------
```

Comment:
Does this mark your new introduced changes? If you so could you elaborated on these changes in your comment?

Reply:
Sure, I've remove this.